### PR TITLE
[Compute] image builder create: Add --vm-size, --os-disk-size, --vnet, --subnet

### PR DESCRIPTION
**Pull request by qwordy**
**Description**  

Resolve https://github.com/Azure/azure-cli/issues/14575

Support vm profile.
```
"vmProfile":
{
"vmSize": "",
"osDiskSizeGB": ,
"vnetConfig": {
"subnetId": "/subscriptions//resourceGroups//providers/Microsoft.Network/virtualNetworks//subnets/"
}
```

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
image builder

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -202,6 +202,12 @@ def load_arguments(self, _):
         c.argument('image_template', help='Local path or URL to an image template file. When using --image-template, all other parameters are ignored except -g and -n. Reference: https://docs.microsoft.com/en-us/azure/virtual-machines/linux/image-builder-json')
         c.argument('identity', nargs='+', help='List of user assigned identities (name or ID, space delimited) of the image template.')
 
+        # VM profile
+        c.argument('vm_size', help='VM size. E.g. Standard_D1_v2')
+        c.argument('os_disk_size', help='Disk size in GB of VM')
+        c.argument('vnet', help='Name of VNET for VM. You should only specify it when subnet is a name')
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:208]: can vnet be a id?

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:208]: We don't need it. If a user prefer ID, vnet should be left empty.

File: src/azure-cli/azure/cli/command_modules/vm/_image_builder.py [hunk 2]
```diff
@@ -377,17 +377,19 @@ def process_img_tmpl_output_add_namespace(cmd, namespace):
 
 # region Custom Commands
 
-def create_image_template(  # pylint: disable=too-many-locals, too-many-branches, too-many-statements
+def create_image_template(  # pylint: disable=too-many-locals, too-many-branches, too-many-statements, unused-argument
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/vm/_image_builder.py:380]: Which argument is not used ? Can we remove the argument directly ?

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_image_builder.py:380]: I didn't add unused argument. Maybe open another PR to address it.

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_image_builder.py:380]: They are used in validator `process_image_template_create_namespace`

**Pull request merged**